File: bench/inst.py

class Foo:
	def bar(self):
		self.inst1 = 0
		self.inst2 = 0
		self.inst3 = 0
		self.inst4 = 0
		i = 0
		while i<10000000:
			self.inst1 += 1
			self.inst2 += 2
			self.inst3 += 3
			self.inst4 += 4
			i += 1
		
		return self.inst1

print Foo().bar()

File: bench/inst.rb
require 'benchmark'

class Foo

	def bar
		@inst1 = 0
		@inst2 = 0
		@inst3 = 0
		@inst4 = 0
		i = 0
		while i<10000000
			@inst1 += 1
			@inst2 += 2
			@inst3 += 3
			@inst4 += 4
			i += 1
		end
		
		return @inst1
	end

end

Benchmark.bm do |x|
  x.report { puts Foo.new.bar }
end